registry/storage/driver/s3-aws/s3.go

```diff
@@ -990,6 +990,10 @@ func (d *driver) doWalk(parentCtx context.Context, objectCount *int64, path, pre
 		}
 
 		for _, file := range objects.Contents {
+			// empty prefixes are listed as objects inside its own prefix.
```

Comment:
can you please add a link to the S3 doc that states this behavior
or
add a test case in `s3_test.go`

Reply:
Please check https://docs.aws.amazon.com/AmazonS3/latest/user-guide/using-folders.html

"The Amazon S3 console implements folder object creation by creating a zero-byte object with the folder prefix and delimiter value as the key. These folder objects don't appear in the console. Otherwise they behave like any other objects and can be viewed and manipulated through the REST API, AWS CLI, and AWS SDKs."

/cc @milosgajdos

Reply:
Thanks, @ricardomaraschini. I think @arkodg probably meant if you could add that link to the comment in the source. Either way, if you could find a way to test it, that would be even more awesome